wolvic/browser/tab_jni.cc

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "wolvic/jni_headers/Tab_jni.h"
+
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_controller.h"
+#include "content/public/browser/web_contents.h"
+#include "url/gurl.h"
+#include "wolvic/wolvic_browser_context.h"
+#include "wolvic/wolvic_content_main_delegate.h"
+
+namespace content {
+
+base::android::ScopedJavaLocalRef<jobject> JNI_Tab_CreateWebContents(
+    JNIEnv* env) {
+  auto* delegate = content::WolvicContentMainDelegate::Get();
+  CHECK(delegate->browser_context() != nullptr);
+
+  WebContents::CreateParams create_params(
+      static_cast<BrowserContext*>(delegate->browser_context()), nullptr);
+  delegate->web_contents_list_.push_back(WebContents::Create(create_params));
+  WebContents* web_contents = delegate->web_contents_list_.back().get();
+
+  NavigationController::LoadURLParams params(GURL("https://google.com"));
```

Comment:
This deserves at least a TODO for moving to a non hard coded path.

Ideally we want to delegate to something else, or just let Java part decide which content it loads by using the java class NavigationController reference.

Reply:
Thanks. TODO is added.